src/apps/config/leader.lua

```diff
@@ -169,8 +169,9 @@ local function path_setter_for_grammar(grammar, path)
       assert(grammar.type == 'struct')
       local tail_id = data.normalize_id(tail_name)
       return function(config, subconfig)
-         getter(config)[tail_id] = subconfig
-         return config
+	 local newconfig = lib.deepcopy(config)
```

Comment:
So! It's really useful to have access to both the old graph and the new graph during whilst handling of the name change. I've added a new function to `core.app` which takes the old name and the new name and performs the rename of an app and does all the necessary checks for such an operation. The problem I was running into was that I needed the old name and the new name to rename the app. I _could_ lookup the name by looking for where `/var/run/snabb/PID/name` points too and then get the basename however rather than poking around in `/var/run/snabb`, I think it's cleaner and more reliable to just look at the old config.

I think the code itself was written like this occurs too, from what I can see in `Leader:update_configuration` it tries to do this:

1. Creates a new modified config based on the changes
2. It gives both old and new to compute the actions needed to be done
3. Sends those to the follower
4. Set the new config as the "current"

However when we do step one it actually changes `self.current_app_graph` and so the new and old configuration at least from the `name` prospective are actually the same. I think it makes a bit more sense to create a new configuration, figure out the changes, ask the lwAFTR to do those changes and then modify the leader's idea of the config.

WDYT?